src/please.go

```diff
@@ -826,9 +826,9 @@ func runPlease(state *core.BuildState, targets []core.BuildLabel) {
 // One can also pass a 'failed' flag which runs the failed tests from last time.
 func testTargets(target core.BuildLabel, args []string, failed bool, resultsFile cli.Filepath) []core.BuildLabel {
 	if failed {
-		targets, args := test.LoadPreviousFailures(string(resultsFile))
+		targets, _ := test.LoadPreviousFailures(string(resultsFile))
```

Comment:
Is this not losing the "just run specific failing test cases" behaviour - it will re-run the whole target, no?

Reply:
So if you have multiple targets it will only re-run the targets that had failures. The problem was that the args returned contain both Testcases and sub-Testcases (as the results file records both) and this was causing it to fail with some ReturnValue error. I'll investigate this further.

Reply:
I've reverted this.